hack/tools.go

```diff
@@ -0,0 +1,22 @@
+// +build tools
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This package imports things required by build scripts, to force `go mod` to see them as dependencies
+package tools
+
+import _ "k8s.io/code-generator"
```

Comment:
Without adding this file `make gen` fails with the below error ...
```
make gen
ls: cannot access '_output/bin/golangci-lint': No such file or directory
./hack/update-generated-conversions.sh
+ go build -o /Users/myuser/projects/go/src/sigs.k8s.io/descheduler/_output/bin/conversion-gen k8s.io/code-generator/cmd/conversion-gen
can't load package: package k8s.io/code-generator/cmd/conversion-gen: cannot find package "." in:
	/Users/myuser/projects/go/src/sigs.k8s.io/descheduler/vendor/k8s.io/code-generator/cmd/conversion-gen
make: *** [gen] Error 1
```

Reply:
I'm fine with this, like we talked about before this also lets us pin a code generator version so we don't get the random go.mod updates